### PR TITLE
feat(types): add TypeScript 5.2 support

### LINKED ISSUE #16414: Typescript 5.2 support
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### Describe the feature you'd like to see implemented

```
node_modules/sequelize/types/transaction.d.ts:121:13 - error TS2395: Individual declarations in merged declaration 'LOCK' must be all exported or all local
```

### Describe why you would like this feature to be added to Sequelize
The latest version of typescript 5.2 was released and sequelize is not compatible

### Is this feature dialect-specific?

- [x] No. This feature is relevant to Sequelize as a whole.
- [ ] Yes. This feature only applies to the following dialect(s): 

### Would you be willing to resolve this issue by submitting a Pull Request?

- [x] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in implementing my feature.

---

*Indicate your interest in the addition of this feature by adding the 👍 reaction. Comments such as "+1" will be removed.*

### PR DESCRIPTION
**Pull request by esetnik**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Introduce fixes for typescript 5.2 compatibility in v6. Fixes #16414

### CODE REVIEW

File: src/transaction.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { Deferrable } from './deferrable';
-import { Logging } from './model';
-import { Sequelize } from './sequelize';
+import { Deferrable } from "./deferrable";
```
Comment (WikiRik) [inline src/transaction.d.ts:1]: Can you undo these changes to the quotes? That reduces the amount of changes

Comment (esetnik) [reply src/transaction.d.ts]: I have added a prettierrc file and those changes were corrected in subsequent versions. Please let me know if you want me to remove it.

**Comment by mat813** [comment]
I feel that the `.prettierrc.json`, `package.json` and `yarn.lock` files should not be touched by this merge request.

**Comment by WikiRik** [comment]
> I feel that the `.prettierrc.json`, `package.json` and `yarn.lock` files should not be touched by this merge request.

Agree about prettier. @types/lodash needs to be updated because it will otherwise give errors with TS 5.2

**Review by WikiRik — COMMENTED** [review]
I forgot to review this. If you remove the prettier file it's good to merge

**Review by WikiRik — APPROVED** [review]
